src/os/ui/filter/op/isfalseop.js

```diff
@@ -0,0 +1,71 @@
+goog.provide('os.ui.filter.op.IsFalse');
+
+goog.require('os.ui.filter.op.Op');
+goog.require('os.xsd.DataType');
+
+
+/**
+ * A 'PropertyIsFalse' operation class.
+ * Based on the OGC Filter Spec
+ *
+ * @extends {os.ui.filter.op.Op}
+ * @constructor
+ */
+os.ui.filter.op.IsFalse = function() {
+  os.ui.filter.op.IsFalse.base(this, 'constructor',
+      'And', 'is false', 'false',
+      [os.xsd.DataType.BOOLEAN, os.xsd.DataType.INTEGER, os.xsd.DataType.STRING],
+      'hint="os.ui.filter.op.IsFalse"', 'Supports false, 0, and "false" (case insensitive)', 'span', true);
+  this.matchHint = 'os.ui.filter.op.IsFalse';
+};
+
+
+goog.inherits(os.ui.filter.op.IsFalse, os.ui.filter.op.Op);
+
+
+/**
+ * Because OpenSphere allows isEmpty() comparator for Boolean type properties, fail null/empty tests
+ *
+ * @inheritDoc
+ */
+os.ui.filter.op.IsFalse.prototype.getEvalExpression = function(v, literal) {
+  return '(!(typeof ' + v + '==="undefined"||' + v + '==null||' + v + '.length==0)&&(' +
```

Comment:
A few things here:
* `prop == null` covers both `null` and `undefined` values, so the `typeof v === 'undefined'` check is redundant. `typeof` is also much slower than a simple value test, so try to avoid it when possible.
* You should be able to simplify this test to `v === false || String(val).toLowerCase() === 'false' || v === 0`, and omit the null/undefined/length checks entirely.

Reply:
I recommend not removing the undefined check from code intended to be run through an eval() -- since any unhandled JS exception may cause all JS to cease running.  Also, exception handling is costly as well.

Try typing "blah === null" in your developer tools command line and verify that it doesn't return true or false -- it throws an exception.

This is relevant, because on many JSON-returning backend technologies, it's possible to prune properties that are null to save bandwidth - so some objects may not have all the properties.

Also, not all browsers and all versions of all browsers treat undefined the same way as the others.  Forcing the undefined check through the "typeof" spec is one way to make all browsers treat it the same way.  At one point, undefined wasn't even necessarily equal to null in all browsers -- thus the "redundant" check of "is undefined OR is null."

Since we're pretty much pushing "latest Chrome" and "latest Firefox" I know it's less of a concern with this team -- so let me know if we push code that utilizes features/behaviors that aren't in IE/Safari and/or less-recent versions of Chrome/Firefox.

Anyway, I'll research a newer/faster way of doing the undefined check without a string comparison.  This was just one way I learned that worked in all browsers without throwing exceptions.

Reply:
All of your points are fair and valid, given either of the following is true:
* The root variable name being tested is not defined in the eval scope.
* The variable name causes an undefined reference exception. For example, `foo = {}` and variable name is `foo.bar.baz`.

OpenSphere assumes that code creating filter functions will protect against both of these. It's entirely possible a leaf property name may not exist on an object (ie, `foo.bar` in the above example), but that will not cause an exception with any of the tests we're doing here.

We make these assumptions because these filter functions could potentially be run against millions of objects, so we value performance with the exchange of putting some burden on the developer to protect against the exceptions you mentioned. If we were to change the core assumptions, we would have to update all of the filter operations to perform the additional undefined variable check.

Reply:
Thanks for the break-down!  I was actually thinking about writing a utility function to do dot and array notation resolution on variables (just like you mentioned -- e.g. utils.isNotNull("foo.bar[lookup.idx].baz")) -- but you're right -- the speed requires a trade-off, and assuming there's at least some kind of check upstream then everything downstream can go faster.